### PR TITLE
Modify rule S3464: LaYC format

### PR DESCRIPTION
**Pull request by antonioaversa**
RSpec: https://sonarsource.github.io/rspec/#/rspec/S3464

### CODE REVIEW

File: rules/S3464/csharp/rule.adoc [hunk 1]
```diff
@@ -1,23 +1,95 @@
...
 
-include::../description.adoc[]
+Recursion is a technique used in computer science to define a problem in terms of the problem itself, usually in terms of a simpler version of the problem itself.
 
-=== Noncompliant code example
+For example the implementation of the generator for the n-th value of the Fibonacci sequence comes naturally from its mathematical definition, when recursion is used:
+
+[source,csharp]
+----
+static int NthFibonacciNumber(int n)
```
Comment (gregory-paidis-sonarsource) [inline rules/S3464/csharp/rule.adoc:9]: I think you can remove `static` to simplify the examples.
Applies to all code snippets.

File: rules/S3464/csharp/rule.adoc [hunk 2]
```diff
@@ -1,23 +1,95 @@
...
+
+Use of recursion is acceptable in methods, like the one above, where you can break out of it. 
+
+[source,csharp]
+----
+static int NthFibonacciNumber(int n)
+{
+    if (n <= 1)
+    {
+        return 1; // Base case => stop recurring
```
Comment (gregory-paidis-sonarsource) [inline rules/S3464/csharp/rule.adoc:46]: (Nitpick, optional)
The arrow seems like a lambda to me, maybe use a colon:

`Base case: ...`

Also, `recurring` feels slightly unnatural, maybe something like:

`(stop|exit) the recursive calls` or `(stop|exit) the recursion`

File: rules/S3464/csharp/rule.adoc [hunk 3]
```diff
@@ -1,23 +1,95 @@
...
+    }
+    // ...
+}
+----
+
+It is also acceptable and makes sense in some type definitions:
+
+[source,csharp]
+----
+class Box<T> : IComparable<Box<T>>
```
Comment (gregory-paidis-sonarsource) [inline rules/S3464/csharp/rule.adoc:56]: Do you need the CRTP _target_ to be generic? Is it semantically different than this, as far as the pattern is concerned:

```csharp
class Box: IComparable<Box> { }
```

I feel like if you try to convey CRTP, having one less generic makes it a bit more readable/understandable.

File: rules/S3464/csharp/rule.adoc [hunk 4]
```diff
@@ -1,23 +1,95 @@
...
+{
+    public int CompareTo(Box<T>? other)
+    {
+        // Compare the two Box<T> instances...
+    }
+}
+----
+
+With types, some invalid recursive definitions are catched by the compiler:
```
Comment (gregory-paidis-sonarsource) [inline rules/S3464/csharp/rule.adoc:65]: "catched" -> "caught"

File: rules/S3464/csharp/rule.adoc [hunk 5]
```diff
@@ -1,23 +1,95 @@
...
+class C2<T> : C2<T>     // Error CS0146: Circular base type dependency
+{
+}
+
+class C2<T> : C2<C2<T>> // Error CS0146: Circular base type dependency
+{
+}
+----
+
+In more complex scenarios, however, the code will compile but execution will result into a `TypeLoadException` if you try to instantiate the class.
```
Comment (gregory-paidis-sonarsource) [inline rules/S3464/csharp/rule.adoc:78]: Either here or in "Resources" tab, you can add this:

- [TypeLoadException](https://learn.microsoft.com/en-us/dotnet/api/system.typeloadexception?view=net-7.0)

Some more suggestions:
- [Recursion - wiki](https://en.wikipedia.org/wiki/Recursion_(computer_science))
- [Clippert's "Curiouser and curiouser"](https://ericlippert.com/2011/02/02/curiouser-and-curiouser/)
- [Cleary's CRGP](https://blog.stephencleary.com/2022/09/modern-csharp-techniques-1-curiously-recurring-generic-pattern.html)

You can also add [wiki's CRTP](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern), but it's basically all about C++

**Review by gregory-paidis-sonarsource — CHANGES_REQUESTED** [review]
Looks very readable, left a couple of suggestions. Consider most stuff for VB as well, since it's very similar.

File: rules/S3464/csharp/rule.adoc (hunk 3 above)
Comment (antonioaversa) [reply rules/S3464/csharp/rule.adoc:56]: Yes, I don't think it is actually needed and it doesn't add value. Removed.

File: rules/S3464/csharp/rule.adoc (hunk 4 above)
Comment (antonioaversa) [reply rules/S3464/csharp/rule.adoc:65]: Nice catch :-)

File: rules/S3464/csharp/rule.adoc [hunk 6]
```diff
@@ -1,22 +1,96 @@
...
```
Comment (gregory-paidis-sonarsource) [inline rules/S3464/csharp/rule.adoc:78]: Consider adding the link for `TypeLoadException` here as well, as we discussed.

**Review by gregory-paidis-sonarsource — CHANGES_REQUESTED** [review]
Some more minor changes suggested.

File: rules/S3464/csharp/rule.adoc (hunk 4 above)
Comment (antonioaversa) [reply rules/S3464/csharp/rule.adoc:65]: whoops, fixed!

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**